LDAP: Make LDAP attribute mapping case-insensitive

markkrj opened the pull request:
Make Grafana's LDAP attribute mapping case-insensitive.
I did a lot of LDAP integrations and never came against this issue, so I think this might be the most common behavior.
Even tools like `ldapsearch` and `ldp.exe` do not care about attribute's letter case.

fix: #11300
fix: #58991

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=58992) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=58992) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=58992) it.</sub>

markkrj committed:
Make LDAP attribute mapping case-insensitive

Before this commit, ldap attribute mapping had to match directory
schema's attribute's letter case. This makes it case-insensitive.

fix: #11300
fix: #58991

markkrj committed:
Add test case with attribute name different from schema's

markkrj committed:
Add fix to getArrayAttribute also and add test with mismatched letter
case.

mgyongyosi commented on pkg/services/ldap/helpers.go:
```diff
@@ -49,7 +50,8 @@ func getArrayAttribute(name string, entry *ldap.Entry) []string {
 	for _, attr := range entry.Attributes {
-		if attr.Name == name && len(attr.Values) > 0 {
+		if strings.EqualFold(attr.Name, name) && len(attr.Values) > 0 {
+			fmt.Println(len(attr.Values))
```
Could you please remove this line?

markkrj replied:
Done 😅

mgyongyosi commented on pkg/services/ldap/helpers.go:
```diff
@@ -1,6 +1,7 @@
 package ldap
 
 import (
+	"fmt"
```
```suggestion
```

markkrj committed:
Update pkg/services/ldap/helpers.go

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

mgyongyosi reviewed (approved):
LGTM. I think it shouldn't cause any problem/breaking change, because the current rule (case-sensitivity) is stricter.

fyi @Jguer

mgyongyosi commented:
Thank you @markkrj for this PR! 🙇  It has been merged. 🎉
